eCourse_backend/users/views.py

```diff
@@ -1,62 +1,257 @@
 @login_required
 @permission_required('users.manage_users', raise_exception=True)
-def delete_user(request, id):
-    user_to_delete = get_object_or_404(User, pk=id)
-    # TODO: lecturers do not have matr_nr
-    matr_nr = user_to_delete.matr_nr
-    name = '{} {}'.format(user_to_delete.first_name, user_to_delete.last_name)
-    user_to_delete.delete()
+def student_list_iframe(request, page=1):
+    if request.method == 'POST':
+        filter_form = StudentFilterForm(
+            request.POST, )
+        if filter_form.is_valid():
+            # Filter
+            students = Student.objects.filter(
+                matr_nr__contains=filter_form['matr_nr'].data,
+                first_name__contains=filter_form['first_name'].data,
+                last_name__contains=filter_form['last_name'].data,
+                username__contains=filter_form['username'].data)
+    else:
+        filter_form = StudentFilterForm()
+        students = Student.objects.all()
+
+    paginator = Paginator(students, 10)
+    page_obj = paginator.get_page(page)
+
     context = {
-        'matr_nr': matr_nr,
-        'name': name,
+        'page_obj': page_obj,
+        'filter_form': filter_form,
     }
-    return render(request, 'users/deleted_user.html', context)
+    return render(request, 'admin/users/iframes/student_list.html', context)
 
 
+# Lecturer list
+
+@xframe_options_exempt
+@login_required
+@permission_required('users.manage_users', raise_exception=True)
+def lecturer_list_iframe(request, page=1):
+    if request.method == 'POST':
+        filter_form = LecturerAndOfficeFilterForm(
+            request.POST, )
+        if filter_form.is_valid():
+            # Filter
+            lecturers = Lecturer.objects.filter(
+                first_name__contains=filter_form['first_name'].data,
+                last_name__contains=filter_form['last_name'].data,
+                username__contains=filter_form['username'].data)
+    else:
+        filter_form = LecturerAndOfficeFilterForm()
+        lecturers = Lecturer.objects.all()
+
+    paginator = Paginator(lecturers, 10)
+    page_obj = paginator.get_page(page)
+
+    context = {
+        'page_obj': page_obj,
+        'filter_form': filter_form,
+    }
+    return render(request, 'admin/users/iframes/lecturer_list.html', context)
+
+
+# Staff and Admin List
+@xframe_options_exempt
 @login_required
 @permission_required('users.manage_users', raise_exception=True)
-def alter_user(request, id):
+def staff_admin_list_iframe(request, page=1):
     if request.method == 'POST':
-        form = UserForm(request.POST)
+        filter_form = LecturerAndOfficeFilterForm(
+            request.POST, )
+        if filter_form.is_valid():
+            # Filter
+            users = Office.objects.filter(
+                first_name__contains=filter_form['first_name'].data,
+                last_name__contains=filter_form['last_name'].data,
+                username__contains=filter_form['username'].data)
```

Comment:
Also please at least provide an Example, I don't know how else it could be done
And also what do you mean by validated data? Since we call the .is_valid() function.